### ISSUE TITLE
mrsprolog codec

### ISSUE THREAD

**Issue by arademaker** [arademaker/delphin#4]
1. I would like to use the ideas from https://homepages.inf.ed.ac.uk/wadler/papers/prettier/prettier.pdf implemented in Lean as Format type.
2. I would like to follow the idea of codecs from https://pydelphin.readthedocs.io/en/latest/api/delphin.codecs.html

**Comment by MarcusGDaniels** [comment]
The ToFormat typeclasses don't seem to be designed to add alternative formats.   One idea is to have type tuples that combine the object with a style descriptor.  If that were done, the Mrs/Basic.lean file should be done in the same way, I think.

**Comment by arademaker** [comment]
Indeed, I still need to clearly understand how to use the Prettyprinter to produce alternative formats. I am rereading the discussion at https://leanprover.zulipchat.com/#narrow/stream/419231-Lean-Together-2024/topic/A.20Pretty.20Expressive.20Printer.20-.20Sorawee.20Porncharoenwase.

**Comment by arademaker** [comment]
In this Haskell Library, the prettyprinter is illustrated with different renders 

https://hackage.haskell.org/package/prettyprinter-1.7.1/docs/Prettyprinter.html
